server/index.js

```diff
@@ -51,6 +51,100 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  const { cartId } = req.session;
+  if (!cartId) {
+    return res.status(200).send([]);
+  }
+  const sql = `
+        select "c"."cartItemId",
+          "c"."price",
+          "p"."productId",
+          "p"."image",
+          "p"."name",
+          "p"."shortDescription"
+      from "cartItems" as "c"
+      join "products" as "p" using ("productId")
+    where "c"."cartId" = $1
+  `;
+  const params = [cartId];
+  db.query(sql, params)
+    .then(response => res.status(200).json(response.rows))
+    .catch(err => next(err));
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const { productId } = req.body;
+  if (productId > 0 && typeof parseInt(productId) === 'number') {
+    const params = [productId];
+    const sql = `
+      select "price"
+      from "products"
+      where "productId" = $1`;
+
+    db.query(sql, params)
+      .then(result => {
+        const products = result.rows;
+
+        if (products.length === 0) {
+          return next(new ClientError('There are no products by that Id.', 400));
```

Comment:
You want to either `throw new ClientError('There are no products by that Id.', 400)` or `return Promise.reject(new ClientError('There are no products by that Id.', 400))`

The way it is, it's possible for a client to provide an invalid `productId` and your server will still try the stuff in the promise chain.